blockchain/types/accountkey/account_key_weighted_multi_sig.go

```diff
@@ -97,19 +111,29 @@ func (a *AccountKeyWeightedMultiSig) Validate(r RoleType, recoveredKeys []*ecdsa
 			continue
 		}
 
+		// if the registered key is included in one of the transaction signatures,
+		// update weightedSum and validSigNum
 		if _, ok := pMap[string(b)]; ok {
 			weightedSum += k.Weight
+			validSigNum++
 		}
 	}
 
-	if weightedSum >= a.Threshold {
-		return true
+	// if isIstanbul is true, check whether invalid signature exists
+	if isIstanbul && validSigNum < len(pMap) {
+		logger.Debug("AccountKeyWeightedMultiSig validation failed and invalid signature exists",
+			"validSigNum", validSigNum, "uniqueSigNum", len(pMap))
+		return false, 0
```

Comment:
Shouldn't it return validSigNum?
```suggestion
		return false, validSigNum
```

Reply:
The code is reverted, thanks.